hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java

```diff
@@ -378,8 +383,9 @@ void checkJumboTransactionBody(TransactionInfo txInfo) throws PreCheckException
         final var allowedJumboHederaFunctionalities = jumboTransactionsConfig.allowedHederaFunctionalities();
 
         if (jumboTxnEnabled
-                && txInfo.serializedTransaction().length() > hederaConfig.transactionMaxBytes()
-                && !allowedJumboHederaFunctionalities.contains(fromPbj(txInfo.functionality()))) {
+                && txInfo.transaction().protobufSize() > hederaConfig.transactionMaxBytes()
+                && !allowedJumboHederaFunctionalities.contains(fromPbj(txInfo.functionality()))
+                && !NON_JUMBO_TRANSACTIONS_BIGGER_THAN_6_KB.contains(txInfo.functionality())) {
             throw new PreCheckException(TRANSACTION_OVERSIZE);
```

Comment:
Seems like this is checked at Ingest for all transactions. I am not sure why we need a new check here.

We could just check if the transaction is allowed jumbo transaction if its greater than 6KB. We can fix in next PR if needed.